### ISSUE TITLE
Standard for checking error/ log messages

### ISSUE THREAD

**Issue by jdkandersson** [canonical/is-charms-contributing-guide#4]
- Checking for the entire log message can mean that the tests become dependent on the wording which may change to make it easier to understand
- Focus on key contents (e.g., `relation` and `joined`)
- Don't check for case sensitivity (e.g. convert to lower case before checking)

**Comment by mthaddon** [comment]
I agree checking for an entire log message can mean your tests need changing if a log message changes, but I also think having an explicit log message in the tests makes it much easier to read the tests, and can potentially catch issues that checking individual words might not. For example "The jenkins relation was not joined due to an incompatible error" and "Jenkins relation joined" would both match if you were only looking for the keywords `relation` and `joined` although the meaning is completely different.

**Comment by jdkandersson** [comment]
It is a good point, it is a trade-off. In practice it helps to pick words or phrases that are unique to the log message being looked for and it also helps to include dynamic elements of the log message (such as the name of a unit) to check that it correctly made it in.

**Comment by seb4stien** [comment]
Closing old issues.
